### ISSUE TITLE
Using AssemblyLoadContext.Default.Resolving event quits with ExecutionEngineException

### ISSUE THREAD

**Issue by jcansdale** [dotnet/corefx#10365]
If the `Default.Resolving` event is used to resolve an assembly, the process quits with a `System.ExecutionEngineException`.

Here is a repro project (run as console app or xUnit test):
[System.Runtime.Loader.Tests.zip](https://github.com/dotnet/corefx/files/385861/System.Runtime.Loader.Tests.zip)

![assemblyloadcontext_resolving](https://cloud.githubusercontent.com/assets/11719160/17173083/8dc5f846-53f0-11e6-8c24-ca71964507e0.png)

I'm stuck on this issue here:
https://github.com/jcansdale/TestDriven.Net-Issues/issues/7

**Comment by jcansdale** [comment]
Here is the test:

```
namespace System.Runtime.Loader.Tests
{
    using IO;
    using System;
    using System.Reflection;
    using System.Runtime.Loader;
    using Xunit;

    public class AssemblyLoadContextTests
    {
        // This can be any assembly that hasn't already been loaded.
        const string relativePath = @"System.Diagnostics.TraceSource\4.0.0\runtimes\win\lib\netstandard1.3\System.Diagnostics.TraceSource.dll";

        static string packagesDir = Environment.ExpandEnvironmentVariables(@"%USERPROFILE%\.nuget\packages");
        static string assemblyPath = Path.Combine(packagesDir, relativePath);

        [Fact]
        public void Load_Resolving_ExpectFullName()
        {
            // If you pre-load the assembly by uncommenting this line, it works fine.
            // AssemblyLoadContext.Default.LoadFromAssemblyPath(assemblyPath);
            var assemblyName = AssemblyLoadContext.GetAssemblyName(assemblyPath);

            AssemblyLoadContext.Default.Resolving += new Resolver(assemblyPath).Resolving;
            var asm = Assembly.Load(assemblyName); // Quits here with 'System.ExecutionEngineException'.

            Assert.Equal(assemblyName.FullName, asm.GetName().FullName);
            Console.WriteLine("Hello?");
        }

        class Resolver
        {
            string resolvePath;
            AssemblyName resolveName;

            public Resolver(string assemblyPath)
            {
                resolveName = AssemblyLoadContext.GetAssemblyName(assemblyPath);
                resolvePath = assemblyPath;
            }

            public Assembly Resolving(AssemblyLoadContext context, AssemblyName assemblyName)
            {
                if (assemblyName.FullName == resolveName.FullName)
                {
                    var assembly = context.LoadFromAssemblyPath(resolvePath);
                    Console.WriteLine("Resolving: " + Path.GetFileNameWithoutExtension(resolvePath) + " -> " + assembly);
                    return assembly;
                }

                return null;
            }
        }
    }
}
```

**Comment by gkhanna79** [comment]
Dup of https://github.com/dotnet/coreclr/issues/5837 that is being considered for 1.0.X.

**Comment by jcansdale** [comment]
@gkhanna79 thanks for the update! Are there build artifacts somewhere I could try? (I'm using Windows).

**Comment by gkhanna79** [comment]
You can download the latest CoreCLR nuget package from dotnet.myget.org (dotnet-core feed) to try the fix.

**Comment by jcansdale** [comment]
@gkhanna79,

Thanks for the tip! I tried downloading the latest `Microsoft.NETCore.App` and `System.Runtime.Loader` from the feed at https://dotnet.myget.org/F/dotnet-core/, but the process is still quitting. :cry: 

Should this be working in "4.0.1-beta-24327-03"?


Here's my project.json:
```
{
  "version": "1.0.0-*",
  "buildOptions": {
    "emitEntryPoint": true
  },

  "dependencies": {
    "dotnet-test-xunit": "2.2.0-preview2-build1029",
    "Microsoft.NETCore.App": "1.0.1-beta-000547-00",
    "System.Runtime.Loader": "4.0.1-beta-24327-03",
    "xunit": "2.2.0-beta2-build3300"
  },

  "runtimes": {
    "win10-x64": {}
  },

  "testRunner": "xunit",

  "frameworks": {
    "netcoreapp1.0": {
      "imports": "dnxcore50"
    }
  }
}

```

**Comment by gkhanna79** [comment]
You need to download runtime.win7-x64.Microsoft.NETCore.Runtime.CoreCLR package and replace your CoreCLR build with it to try out the fix.

**Comment by jcansdale** [comment]
I copied the contents of `...\packages\runtime.win7-x64.Microsoft.NETCore.Runtime.CoreCLR\1.0.4-beta-24327-05\runtimes\win7-x64\` (`native` and `lib\netstandard1.0`) into `\Program Files\dotnet\shared\Microsoft.NETCore.App\1.0.0`. Everything seems to work! :+1:

**Comment by gkhanna79** [comment]
This has been fixed in 1.0.1.
